File: week1/problems/mario.c
#include <stdio.h>
#include <cs50.h>

void printChar(string character, int times);

int main(void)
{
    // Ask user for height (1 to 8 inclusive)
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    int hash = 1;
    int space = height - 1;

    for (int i = 0; i < height; i++)
    {
        // Print left pyramid
        printChar(" ", space);
        printChar("#", hash);

        // Gap between two pyramids
        printf("  ");

        // Print right pyramid
        printChar("#", hash);

        // Sequence for hash and space
        hash++;
        space--;

        // Go to next row
        printf("\n");
    }
}

void printChar(string character, int times)
{
    for (int i = 0; i < times; i++)
    {
        printf("%s", character);
    }
}

File: week6/problems/mario.py
from cs50 import get_int


def main():
    height = get_height()

    for i in range(1, height + 1):

        # Print left pyramid
        build(" ", height - i)
        build("#", i)

        # Print middle gap
        build(" ", 2)

        # Print right pyramid 
        build("#", i)
        print()


# Get height from user between 1 and 8 inclusive
def get_height():
    while True:
        n = get_int("Height: ")
        if n > 0 and n < 9:
            break
    return n


# Print characters n times in one line
def build(char, n):
    for i in range(n):
        print(char, end="")


if __name__ == "__main__":
    main()